src/components/ListView.jsx

```diff
@@ -15,25 +27,59 @@ function ListView() {
       navigate('/');
       return;
     }
+
     const unsubscribe = onSnapshot(collection(db, localToken), (snapshot) => {
       const snapshotDocs = [];
-      snapshot.forEach((doc) => snapshotDocs.push(doc.data()));
+      snapshot.forEach((doc) => {
+        snapshotDocs.push({ ...doc.data(), id: doc.id });
+      });
       setItems(snapshotDocs);
     });
     return () => {
       //Used to remove the snapshot listener when the component is unmounted/unsubscribed
       unsubscribe();
     };
-  }, []);
+  }, [localToken, navigate]);
+
+  //
+  const handleCheckboxChange = async (e) => {
+    const itemId = e.target.name;
+    //create time variable to save time when user checked the box as purchasedTime
+    const datePurchased = Date.now();
+    //if user want to uncheck the item it can be done and purchasedDate is set to null again
+    if (e.target.checked) {
+      setUpdateToDb(localToken, itemId, 'purchasedDate', datePurchased);
+    } else {
+      setUpdateToDb(localToken, itemId, 'purchasedDate', null);
+    }
+  };
+
+  //persists checked box for 24 hours
+  function within24hours(date) {
+    let timeCheck = false;
+    const gap = currentTime - date;
+    if (gap < hours24gap) {
+      timeCheck = true;
+      return timeCheck;
```

Comment:
Tiniest comment - I think you can remove the `return timeCheck` inside the if statement.